src/fixate/drivers/dmm/fluke_8846a.py

```diff
@@ -268,10 +273,54 @@ def voltage_dc(self, _range=None, auto_impedance=False):
             command = "; :SENS:VOLT:DC:IMP:AUTO OFF"
         self._set_measurement_mode("voltage_dc", _range, suffix=command)
 
-    def current_ac(self, _range=None):
+    def current_ac(self, _range, port):
+        """
+        Set the measurement mode on the DMM to AC current.
+
+        If the range and port selection are not compatible, i.e. someone has requested to measure
+        1 A on the low range port with a maximum capability of 400 mA, an exception is raised.
+
+        If the range requested can be measured by the low port, but the high port is selected, an
+        exception is raised.
+        """
+
+        # Check the requested range is not more than the port capability:
+        if _range > self.current_ports[port]:
+            raise ValueError(
+                "The selected port and range combination is not available for this instrument. Consider using a different multimeter"
+            )
+
+        # Raise an error if the high port is selected when the low port should be used:
+        if _range < self.current_ports["LOW"] and port == "HIGH":
+            raise ValueError(
+                "High range port selected when the low range port should be used! Consider using a different multimeter."
+            )
+
         self._set_measurement_mode("current_ac", _range)
 
-    def current_dc(self, _range=None):
+    def current_dc(self, _range, port: Literal["HIGH", "LOW"]):
+        """
+        Set the measurement mode on the DMM to DC current.
+
+        If the range and port selection are not compatible, i.e. someone has requested to measure
+        1A on the low range port with a maximum capability of 400 mA, an exception is raised.
+
+        If the range requested can be measured by the low port, but the high port is selected, an
+        exception is raised.
+        """
+
+        # Check the requested range is not more than the port capability:
+        if _range > self.current_ports[port]:
```

Comment:
Instead of duplicating these checks, could we pull them out into their own method and calling that from the AC and DC methods?

Reply:
Moved the checks over to a function that is now defined in the helper. That way both the drivers just use that function.